tests/test_contact.py

```diff
@@ -5,108 +5,92 @@
-        con = telegram.Contact.de_json(self.json_dict, self._bot)
-        message = self._bot.send_contact(contact=con, chat_id=self._chat_id)
-        contact = message.contact
+    def test_send_with_contact(self, monkeypatch, bot, chat_id, contact):
+        def test(_, url, data, **kwargs):
+            phone = data['phone_number'] == contact.phone_number
+            first = data['first_name'] == contact.first_name
+            last = data['last_name'] == contact.last_name
+            return phone and first and last
 
-        self.assertEqual(contact, con)
-    '''
+        monkeypatch.setattr('telegram.utils.request.Request.post', test)
+        message = bot.send_contact(contact=contact, chat_id=chat_id)
+        assert message
 
-    def test_contact_to_json(self):
-        contact = telegram.Contact.de_json(self.json_dict, self._bot)
+    def test_send_contact_without_required(self, bot, chat_id):
+        with pytest.raises(ValueError, match='Either contact or phone_number and first_name'):
+            bot.send_contact(chat_id=chat_id)
 
-        self.assertTrue(self.is_json(contact.to_json()))
+    def test_to_dict(self, contact):
+        contact_dict = contact.to_dict()
 
-    def test_contact_to_dict(self):
-        contact = telegram.Contact.de_json(self.json_dict, self._bot)
-
-        self.assertTrue(self.is_dict(contact.to_dict()))
-        self.assertEqual(contact['phone_number'], self.phone_number)
-        self.assertEqual(contact['first_name'], self.first_name)
-        self.assertEqual(contact['last_name'], self.last_name)
-        self.assertEqual(contact['user_id'], self.user_id)
+        assert isinstance(contact_dict, dict)
+        assert contact_dict['phone_number'] == contact.phone_number
+        assert contact_dict['first_name'] == contact.first_name
+        assert contact_dict['last_name'] == contact.last_name
+        assert contact_dict['user_id'] == contact.user_id
 
     def test_equality(self):
-        a = telegram.Contact(self.phone_number, self.first_name)
-        b = telegram.Contact(self.phone_number, self.first_name)
-        c = telegram.Contact(self.phone_number, "")
-        d = telegram.Contact("", self.first_name)
-        e = telegram.Voice("", 0)
-
-        self.assertEqual(a, b)
-        self.assertEqual(hash(a), hash(b))
-        self.assertIsNot(a, b)
-
-        self.assertEqual(a, c)
-        self.assertEqual(hash(a), hash(c))
-
-        self.assertNotEqual(a, d)
-        self.assertNotEqual(hash(a), hash(d))
-
-        self.assertNotEqual(a, e)
-        self.assertNotEqual(hash(a), hash(e))
+        a = Contact(self.phone_number, self.first_name)
+        b = Contact(self.phone_number, self.first_name)
+        c = Contact(self.phone_number, '')
+        d = Contact('', self.first_name)
+        e = Voice('', 0)
 
+        assert a == b
+        assert hash(a) == hash(b)
+        assert a is not b
 
-''' Commented out, because it would cause "Too Many Requests (429)" errors.
```

Comment:
It's marked as xfail if it raises RetryError or if on pypy (where it's strangely more unstable)